### PR TITLE
Cohere bench improvements.

### PR DESCRIPTION
**Pull request by afoucret**

### CODE REVIEW

File: cohere_vector/challenges/default.json [hunk 1]
```diff
@@ -44,16 +44,38 @@
...
+    },
+    {
+      "parallel": {
+        "tasks": [
+          {
+            "name": "parallel-documents-indexing-bulk",
+            "operation": "parallel-documents-indexing",
+            "clients": {{ parallel_indexing_bulk_clients | default(1) | int }},
+            "time-period": {{ parallel_indexing_time_period | default(1800) | int }},
+            "target-throughput": {{ parallel_indexing_bulk_target_throughput | default(1) | int }}
```
Comment (jimczi) [inline cohere_vector/challenges/default.json:68]: I guess we also need the bulk size to derive the throughput in terms of docs per second?

File: cohere_vector/challenges/default.json [hunk 2]
```diff
@@ -44,16 +44,38 @@
...
+      "iterations": {{ standalone_search_iterations | default(10000) | int }}
     },
     {
-      "name": "knn-search-100-1000",
+      "name": "standalone-search-knn-100-1000",
       "operation": "knn-search-100-1000",
       "warmup-iterations": 100,
-      "iterations": 1000
+      "clients": {{ standalone_search_clients | default(1) | int }},
+      "iterations": {{ standalone_search_iterations | default(10000) | int }}
```
Comment (jimczi) [inline cohere_vector/challenges/default.json:58]: Can we also another task to test with 8 clients? Same for knn-10-100.

File: cohere_vector/index-vectors-only-mapping.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+  "settings": {
+    "index": {
+      "number_of_shards": {{number_of_shards | default(1)}},
+      "number_of_replicas": {{number_of_replicas | default(0)}}
```
Comment (jimczi) [inline cohere_vector/index-vectors-only-mapping.json:5]: Should we add the preloading option too:

```
"store.preload": ["vec", "vex", "vem"]
```

**Review by jimczi — APPROVED** [review]
I left some comments on the missing bits that we discussed but the change looks great.

File: cohere_vector/challenges/default.json (hunk 2 above)
Comment (afoucret) [reply cohere_vector/challenges/default.json:71]: We now have two flavor of the knn searches`*single-client`and `*-multiple-clients` For multiple clients, I kept the number of clients configurable with `8` as a default.

File: cohere_vector/challenges/default.json (hunk 1 above)
Comment (afoucret) [reply cohere_vector/challenges/default.json:81]: This is right.

**Pull request merged**